lib/beaker/host/unix/exec.rb

```diff
@@ -142,7 +142,7 @@ def delete_env_var key, val
   #  host.get_env_var('path')
   def get_env_var key
     key = key.to_s
-    exec(Beaker::Command.new("env | grep #{key}"), :accept_all_exit_codes => true).stdout.chomp
+    exec(Beaker::Command.new("env | grep ^#{key}="), :accept_all_exit_codes => true).stdout.chomp
```

Comment:
I also wouldn't worry about the injection in this case.  If you have ability to exec arbitrary commands already on a host that you have root / Administrator access to - is there really any increased risk?